guides/theming.md

```diff
@@ -1,311 +1,359 @@
+[Palettes](#palettes) section above, as well as four optional hue numbers. These four hues
+represent, in order: the "default" hue, a "lighter" hue, a "darker" hue, and a "text" hue.
+Components use these hues to choose the most appropriate color for different parts of
+themselves.
+
+```scss
+@use '~@angular/material' as mat;
+
+$my-primary: mat.define-palette(mat.$indigo-palette, 500);
+$my-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
+
+// The "warn" palette is optional and defaults to red if not specified.
+$my-warn: mat.define-palette(mat.$red-palette);
 ```
-Then include the output file in your index.html.
 
-Your custom theme file **should not** be imported into other SCSS files. This will duplicate styles
-in your CSS output. If you want to consume your theme definition object
-(e.g., `$candy-app-theme`) in other SCSS files, then the definition of the theme object should be
-broken into its own file, separate from the inclusion of the `mat-core` and
-`angular-material-theme` mixins.
+You can construct a theme by calling either `define-light-theme` or `define-dark-theme` with
+the result from `define-palette`. The choice of a light versus a dark theme determines the
+background and foreground colors used throughout the components.
+
+```scss
+@use '~@angular/material' as mat;
 
-The theme file can be concatenated and minified with the rest of the application's css.
+$my-primary: mat.define-palette(mat.$indigo-palette, 500);
+$my-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
 
-Note that if you include the generated theme file in the `styleUrls` of an Angular component, those
-styles will be subject to that component's [view encapsulation](https://angular.io/docs/ts/latest/guide/component-styles.html#!#view-encapsulation).
+// The "warn" palette is optional and defaults to red if not specified.
+$my-accent: mat.define-palette(mat.$red-palette);
+
+$my-theme: mat.define-light-theme((
+ color: (
+   primary: $my-primary,
+   accent: $my-accent,
+   warn: $my-warn,
+ )
+));
+```
 
-### Multiple themes
+#### Applying a theme to components
 
-You can create multiple themes for your application by including the `angular-material-theme` mixin
-multiple times, where each inclusion is gated by an additional CSS class.
+The `core-theme` Sass mixin emits prerequisite styles for common features used by multiple
+components. This mixin must be included once per theme.
 
-Remember to only ever include the `@mat-core` mixin once; it should not be included for each
-theme.
+Each Angular Material component has a "color" mixin that emits the component's color styles and
+a "typography" mixin that emits the component's typography styles.
 
-#### Example of defining multiple themes:
+Additionally, each component has a "theme" mixin that emits styles for both color and typography if
+they are specified in the theme defined by `define-light-theme` or `define-dark-theme`. Apply the
```

Comment:
Not sure you need to repeat the light/dark part here since the emphasis is that you can override generic `core-theme` with individual component's `<component-name>-theme` definitions.

Reply:
I reworded this to try to be more clear. The main thing I'm trying to convey here is that if you don't specify e.g. a typography config, the `-theme` mixin won't emit typography styles (null config === no styles).